slim_trees/sklearn_tree.py

```diff
@@ -141,7 +141,10 @@ def _decompress_tree_state(state: dict):
             "values",
         },
         *({"missing_go_to_left"} if sklearn_version >= Version("1.3") else set()),
-    }
+    }:
+        raise ValueError(
+            "Invalid tree structure. Do you use an unsupported scikit-learn version?"
+        )
```

Comment:
A failure can also occur for supported scikit-learn versions. Namely, when unpickling a 1.1 model with version 1.3. Can we add support for that?

Reply:
Would it also fail in this `state.keys() != ...` part or do you want an additional check if the scikit-learn versions don't match that a `ValueError` gets thrown?
I think scikit-learn already warns you if you want to unpickle a model with a newer version than it was pickled...

Reply:
Yes, `sklearn` does issue a warning, but with a custom `Unpickler`, it works.

However, one would get stuck here.